### DISCUSSION TITLE
Do some options to remain some boundary unchange?

### DISCUSSION THREAD

**Discussion by z-g-h** [mdolab/dafoam#725] [General]
Hi, professor
I am using myself couple boundary to optimize multistage turbomachinery. My geometry and FFD box are there (Red line is myself mixing-Plane boundary).  When I tried to run this case, I only wanted to deform the shape of the blade, but I found that the mixing plane boundary also deforms. How should I set it up to prevent deformation on this boundary? Thanks for your help!
![Screenshot 2024-12-20 100213](https://github.com/user-attachments/assets/83fe21df-a128-4e73-b02b-3502ac04d100)
![deform](https://github.com/user-attachments/assets/624c2474-566e-439a-89e4-3c251d71ebb0)

**Comment by friedenhe** [comment]
You can add `or bc["type"] == "mixingPlane"` to here  https://github.com/mdolab/dafoam/blob/08422e96df249433b607455f451d193d21f1b91e/dafoam/pyDAFoam.py#L3671. By default, we don't change wall and cyclic BC types if they are not within the FFD box. You can extend it for the mixingPlan type.
